components/tabs/src/sync/engine.rs

```diff
@@ -144,9 +148,19 @@ impl<'a> SyncEngine for TabsEngine<'a> {
             };
             remote_tabs.push(tab);
         }
-        self.storage.replace_remote_tabs(remote_tabs);
+        self.store
+            .storage
+            .lock()
+            .unwrap()
+            .replace_remote_tabs(remote_tabs);
         let mut outgoing = OutgoingChangeset::new("tabs", inbound.timestamp);
```

Comment:
I think we should move the creation of outgoing up a statement, then just take the mutex once to give it the remote ones and to fetch the local ones.